Source/WTF/wtf/SystemTracing.h

```diff
@@ -173,10 +180,445 @@ enum TracePointCode {
+        case InitializeWebProcessEnd:
+            return "InitializeWebProcess";
+        case RenderingUpdateRunLoopObserverStart:
+        case RenderingUpdateRunLoopObserverEnd:
+            return "RenderingUpdateRunLoopObserver";
+        case LayerTreeFreezeStart:
+        case LayerTreeFreezeEnd:
+            return "LayerTreeFreeze";
+        case FlushRemoteImageBufferStart:
+        case FlushRemoteImageBufferEnd:
+            return "FlushRemoteImageBuffer";
+        case CreateInjectedBundleStart:
+        case CreateInjectedBundleEnd:
+            return "CreateInjectedBundle";
+        case PaintSnapshotStart:
+        case PaintSnapshotEnd:
+            return "PaintSnapshot";
+        case RenderServerSnapshotStart:
+        case RenderServerSnapshotEnd:
+            return "RenderServerSnapshot";
+        case TakeSnapshotStart:
+        case TakeSnapshotEnd:
+            return "TakeSnapshot";
+        case SyntheticMomentumStart:
+        case SyntheticMomentumEnd:
+            return "SyntheticMomentum";
+        case SyntheticMomentumEvent:
+            return "SyntheticMomentumEvent";
+        case RemoteLayerTreeScheduleRenderingUpdate:
+            return "RemoteLayerTreeScheduleRenderingUpdate";
+
+        case CommitLayerTreeStart:
+        case CommitLayerTreeEnd:
+            return "CommitLayerTree";
+        case ProcessLaunchStart:
+        case ProcessLaunchEnd:
+            return "ProcessLaunch";
+        case InitializeSandboxStart:
+        case InitializeSandboxEnd:
+            return "InitializeSandbox";
+        case WebXRCPFrameWaitStart:
+        case WebXRCPFrameWaitEnd:
+            return "WebXRCPFrameWait";
+        case WebXRCPFrameStartSubmissionStart:
+        case WebXRCPFrameStartSubmissionEnd:
+            return "WebXRCPFrameStartSubmission";
+        case WebXRCPFrameEndSubmissionStart:
+        case WebXRCPFrameEndSubmissionEnd:
+            return "WebXRCPFrameEndSubmission";
+
+        case WakeUpAndApplyDisplayListStart:
+        case WakeUpAndApplyDisplayListEnd:
+            return "WakeUpAndApplyDisplayList";
+
+
+        case WTFRange:
+        case JavaScriptRange:
+        case WebCoreRange:
+        case WebKitRange:
+        case WebKit2Range:
+        case UIProcessRange:
+        case GPUProcessRange:
+            return nullptr;
+        }
+    }
+
+    std::string processName { "WebKit" };
+    static const char* getProcessName()
+    {
+        return getInstance().processName.c_str();
+    }
+
+    SysprofAnnotator() { };
+    ~SysprofAnnotator()
+    {
+        for (auto const & iter : ongoingMarks) {
+            std::pair<const void*, const void*> key = iter.first;
+            const void* pointer = std::get<0>(key);
+            const char* name = static_cast<const char*>(std::get<1>(key));
+            endMark(pointer, name, nullptr);
+        }
+    }
+
+    SysprofAnnotator(SysprofAnnotator const&) = delete;
+    void operator=(SysprofAnnotator const&) = delete;
```

Comment:
You could replace those with `WTF_MAKE_NONCOPYABLE()`, as in:

```suggestion
    WTF_MAKE_NONCOPYABLE(SysprofAnnotator);
```